src/main/java/teammates/ui/controller/InstructorCourseInstructorDeleteAction.java

```diff
@@ -27,6 +28,15 @@ protected ActionResult execute() {
 
         /* Process deleting an instructor and setup status to be shown to user and admin */
         if (hasAlternativeInstructor(courseId, instructorEmail)) {
+            List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+            isError = !logic.isAtLeastOneInstructorVisibleToStudents(instructors, instructorEmail);
```

Comment:
You are right, but look at the way it is done: fetch all instructors then check for value of attributes. That is not cost-friendly. It should be able to be incorporated into SQL query, something like `SELECT FROM Instructor WHERE isDisplayedToStudents == true && email != emailParam PARAMETERS String emailParam`.